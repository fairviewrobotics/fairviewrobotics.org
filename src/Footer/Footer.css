.social-media {
  height: 10vh;
  width: 30vw;
}

.absolute {
  position: absolute;
  left: 50%;
  transform: translateX(-50%);
  bottom: 2em;
}

.inline {
  margin-left: auto;
  margin-right: auto;
  margin-bottom: 5em;
}

.social-email {
  font-size: 1em;
  text-align: center;
  right: 0;
  left: 0;
  margin-left: auto;
  margin-right: auto;
  margin-top: 9vh;
  position: absolute;
  color: var(--main-color);
}

.social-email:hover {
  color: var(--second-color);
}

@media(max-width:768px) {
  .social-media {
    width: 80vw;
  }
}
