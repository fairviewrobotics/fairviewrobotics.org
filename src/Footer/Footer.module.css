.socialMedia {
  height: 10vh;
  width: 30vw;
}

/* TODO: use flex instead */
.absolute {
  position: absolute;
  left: 50%;
  transform: translateX(-50%);
  bottom: 2em;
}

.inline {
  margin-left: auto;
  margin-right: auto;
  margin-bottom: 5em;
}

.email {
  font-size: 1em;
  text-align: center;
  right: 0;
  left: 0;
  margin-left: auto;
  margin-right: auto;
  margin-top: 9vh;
  position: absolute;
  color: var(--main-color);
}

.email:hover {
  color: var(--second-color);
}

@media(max-width:768px) {
  .socialMedia {
    width: 80vw;
  }
}
