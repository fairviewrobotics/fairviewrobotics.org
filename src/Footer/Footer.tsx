import React, { PureComponent } from 'react';
import SocialIconList from "./SocialIconList/SocialIconList";

import styles from './Footer.module.css';

export type FooterProps = {
  socialMedia: { url: string, icon: any }[];
  fixed: boolean;
}

export default class Footer extends PureComponent<FooterProps> {

  static defaultProps = {
    fixed: false
  };

  render() {
    const footerType = this.props.fixed ? styles.absolute : styles.inline;

    return (
      <div className={`${styles.socialMedia} ${footerType}`}>
        <SocialIconList socialMedia={this.props.socialMedia}/>
        <a className={styles.email} href="mailto:fairviewrobotics@gmail.com">FAIRVIEWROBOTICS@GMAIL.COM</a>
      </div>
    );
  }
}
