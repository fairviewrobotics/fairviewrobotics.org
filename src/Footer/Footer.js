import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import SocialIconList from "./SocialIconList/SocialIconList";

import './Footer.css'

export default class Footer extends PureComponent {

  static propTypes = {
    socialMedia: PropTypes.arrayOf(PropTypes.shape({
      url: PropTypes.string.isRequired,
      icon: PropTypes.object.isRequired
    })).isRequired,
    fixed: PropTypes.bool.isRequired
  };

  render() {
    const footerType = this.props.fixed ? 'absolute' : 'inline';

    return (
      <div className={`social-media ${footerType}`}>
        <SocialIconList socialMedia={this.props.socialMedia}/>
        <a className="social-email" href="mailto:fairviewrobotics@gmail.com">FAIRVIEWROBOTICS@GMAIL.COM</a>
      </div>
    );
  }
}
