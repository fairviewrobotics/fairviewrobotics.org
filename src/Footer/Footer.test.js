import React from "react";
import { shallow } from "enzyme";
import Footer from "./Footer";

describe('Footer', () => {
  let wrapper;

  it('renders without crashing', () => {
    shallow(
      <Footer socialMedia={[]} />
    );
  });

  describe('given `socialMedia`', () => {
    beforeEach(() => {
      const socialMedia = [
        {
          icon: {
            name: 'githubIcon'
          },
          url: 'http://github.com'
        },
        {
          icon: {
            name: 'facebookIcon'
          },
          url: 'http://facebook.com'
        },
        {
          icon: {
            name: 'twitterIcon'
          },
          url: 'http://twitter.com'
        }
      ];

      wrapper = shallow(
        <Footer socialMedia={socialMedia}/>
      );
    });

    it('should render a inline social media list and an email', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('when `fixed`', () => {
    beforeEach(() => {
      const socialMedia = [];
      wrapper = shallow(
        <Footer fixed socialMedia={socialMedia}/>
      );
    });

    it('should render an absolute positioned list', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});