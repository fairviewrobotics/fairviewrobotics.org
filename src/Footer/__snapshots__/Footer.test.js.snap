// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer given \`socialMedia\` should render a inline social media list and an email 1`] = `
<div
  className="socialMedia inline"
>
  <SocialIconList
    socialMedia={
      Array [
        Object {
          "icon": Object {
            "name": "githubIcon",
          },
          "url": "http://github.com",
        },
        Object {
          "icon": Object {
            "name": "facebookIcon",
          },
          "url": "http://facebook.com",
        },
        Object {
          "icon": Object {
            "name": "twitterIcon",
          },
          "url": "http://twitter.com",
        },
      ]
    }
  />
  <a
    className="email"
    href="mailto:fairviewrobotics@gmail.com"
  >
    FAIRVIEWROBOTICS@GMAIL.COM
  </a>
</div>
`;

exports[`Footer given \`socialMedia\` should render a social media list and an email 1`] = `
<div
  className="socialMedia inline"
>
  <SocialIconList
    socialMedia={
      Array [
        Object {
          "icon": Object {
            "name": "githubIcon",
          },
          "url": "http://github.com",
        },
        Object {
          "icon": Object {
            "name": "facebookIcon",
          },
          "url": "http://facebook.com",
        },
        Object {
          "icon": Object {
            "name": "twitterIcon",
          },
          "url": "http://twitter.com",
        },
      ]
    }
  />
  <a
    className="email"
    href="mailto:fairviewrobotics@gmail.com"
  >
    FAIRVIEWROBOTICS@GMAIL.COM
  </a>
</div>
`;

exports[`Footer when \`fixed\` should render an absolute positioned list 1`] = `
<div
  className="socialMedia absolute"
>
  <SocialIconList
    socialMedia={Array []}
  />
  <a
    className="email"
    href="mailto:fairviewrobotics@gmail.com"
  >
    FAIRVIEWROBOTICS@GMAIL.COM
  </a>
</div>
`;
